File: R-funcs/太乙神数.R
# 将公元纪年数转换为太乙积年数
TaiYiJiNianShu <- function(x){
  
  # 如果输入公元前纪年为负，则输出加1后加上10153977的结果
  if (x < 0) {
    x <- x + 1 + 10153977
  } else {
    # 输入公元后纪年数值为正，则直接加上10153977的结果
    x <- x + 10153977
  }
  
  # 返回结果
  return(x)
}
TaiYiJiNianShu(-1)
TaiYiJiNianShu(0)
TaiYiJiNianShu(2023)

TaiYiGongShu <- function(x){
  result <- x %% 24
  return(list(DeShu=floor(result/3), YuShu=result%%3))
}

TaiYiGongShu(TaiYiJiNianShu(2023))

File: R-funcs/太乙神数.py
# 定义函数
def TaiYiJiNianShu(num):
    # 如果输入数字为负数，则加1后加上10153977
    if num < 0:
        result = num + 1 + 10153977
    else:
        result = num + 10153977
    return result

def TaiYiGongShu(num):
    result = num % 24
    return [int(result/3), result%3]

# 获取用户输入
input_num = int(input("请输入一个数字："))
# 调用函数并输出结果
print("输出结果：", TaiYiJiNianShu(input_num))
print("输出结果：", TaiYiGongShu(TaiYiJiNianShu(input_num)))
